### PR TITLE
Introduction of gcc-11 to CI

### PR DESCRIPTION
**Pull request by pwisniewskimobica**
Switch to gcc11 and g++11

### CODE REVIEW

**Comment by pwisniewskimobica** [comment]
We want to introduce compilation with gcc-11 in Flang CI.
This PR lets us build the gcc-11 artifacts for later use in Flang repository CI which is done in
flang-compiler/flang#1190

Kind request for your reviews, @bryanpkc , @shivaramaarao , @mleair , @kiranchandramohan
Thank you in advance!

**Review by bryanpkc — APPROVED** [review]
LGTM

**Comment by pawosm-arm** [comment]
What about the merge commit this PR contains? Are we happy with that @bryanpkc ?

**Review by bryanpkc — CHANGES_REQUESTED** [review]
@pawosm-arm Thanks for catching the merge commit! I hadn't noticed it since the diffs looked identical to #80, and that one seemed fine from a commit perspective.

@pwisniewskimobica Could you please rebase this branch and remove the merge commit?

**Comment by pwisniewskimobica** [comment]
@bryanpkc 
@pawosm-arm 
rebase applied
-->fixed

File: .github/workflows/pre-compile_llvm.yml [hunk 1]
```diff
@@ -3,26 +3,23 @@ name: Pre-compile llvm
 on:
   workflow_dispatch:
   push:
-    branches: [ release_100, release_11x, release_12x, release_13x ]
+    branches: [ release_100, release_11x, release_12x, release_13x]
```
Comment (bryanpkc) [inline .github/workflows/pre-compile_llvm.yml]: Unnecessary diff. If you insist on removing the space, you should do the same after the left bracket, and also make sure that the formatting is the same in the other two PRs.

File: .github/workflows/pre-compile_llvm.yml [hunk 2]
```diff
@@ -3,26 +3,23 @@ name: Pre-compile llvm
...
 
 jobs:
   build:
+    env:
+      cpp: "g++"
```
Comment (bryanpkc) [inline .github/workflows/pre-compile_llvm.yml]: Remove this and use `echo "cpp=g++" >> $GITHUB_ENV`in the appropriate steps below.

**Comment by bryanpkc** [comment]
I would suggest rewording the commit to something like:
```
Use GCC 11 in CI builds
```

File: .github/workflows/pre-compile_llvm.yml (hunk 1 above)
Comment (pwisniewskimobica) [reply .github/workflows/pre-compile_llvm.yml]: fixed

File: .github/workflows/pre-compile_llvm.yml (hunk 2 above)
Comment (pwisniewskimobica) [reply .github/workflows/pre-compile_llvm.yml]: fixed

**Comment by bryanpkc** [comment]
Could you remove the word "fix" in the commit message?

**Comment by pwisniewskimobica** [comment]
@bryanpkc redundant word 'fix removed
-->fixed

**Review by pawosm-arm — APPROVED** [review]
